src/Backend/Synapse/Importer.php

```diff
@@ -241,17 +237,108 @@ private function doNonIncrementalLoad(
         SynapseImportOptions $importOptions,
         Storage\SourceInterface $source,
         Storage\Synapse\Table $destination,
-        array $primaryKeys
+        DestinationTableOptions $destinationOptions
+    ): void {
+        if (!empty($destinationOptions->getPrimaryKeys())) {
+            if ($importOptions->useOptimizedDedup()) {
+                $this->doFullLoadWithDedup(
+                    $source,
+                    $destination,
+                    $importOptions,
+                    $destinationOptions
+                );
+            } else {
+                $this->doLegacyFullLoadWithDedup(
+                    $source,
+                    $destination,
+                    $importOptions,
+                    $destinationOptions
+                );
+            }
+            return;
+        }
+
+        $this->doLoadFullWithoutDedup(
+            $source,
+            $destination,
+            $importOptions
+        );
+    }
+
+    private function doFullLoadWithDedup(
```

Comment:
Ostatní loady ale běží v transakci. Tady dropneš cílovou tabulku a pak importuješ data

- když se vydrbe import tak jsi přišel o původní tabulku
- když ten import pobězí dlouho (jako že poběží) tak budou failovat další operace které s tou tabulkou pracují protože nebude existovat

Reply:
A máš nějaké řešení? Možeme to nějak probrat s Padákem, protože já řešení jiné moc nevidím :/

Reply:
Vytvořit to "bokem" a pak tou tabulkou nahradit tu původní? (replace/rename)

Reply:
hah to mě nenapadlo, přitom je to zrovna v Docs trošku sa stydím. Už sem to upravil.

Reply:
@zajca V jaky docs?

Reply:
https://docs.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/sql-data-warehouse-develop-ctas#use-ctas-to-copy-a-table